From d9d2159b7f992eb0c8f9434e9943cff34640773f Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Tue, 23 Jan 2018 13:29:19 +0800
Subject: [PATCH 3241/5242] MLK-16224-3: ASoC: imx-pcm-dma-v2: fifo_num is
 used by dma slave config

commit  82a23a6c67a981c50b58a718483c1510ef4c2c3a from
https://source.codeaurora.org/external/imx/linux-imx.git

fifo_num is a new added parameter for dma slave config

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Viorel Suman <viorel.suman@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/fsl/imx-pcm-dma-v2.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/sound/soc/fsl/imx-pcm-dma-v2.c b/sound/soc/fsl/imx-pcm-dma-v2.c
index 2114c4a5..470e85a 100644
--- a/sound/soc/fsl/imx-pcm-dma-v2.c
+++ b/sound/soc/fsl/imx-pcm-dma-v2.c
@@ -82,6 +82,11 @@ static int imx_pcm_hw_params(struct snd_pcm_substream *substream,
 					dma_data,
 					&config);
 
+	if (substream->stream == SNDRV_PCM_STREAM_PLAYBACK)
+		config.dst_fifo_num = dma_data->fifo_num;
+	else
+		config.src_fifo_num = dma_data->fifo_num;
+
 	return dmaengine_slave_config(chan, &config);
 }
 
-- 
1.7.9.5

